### DISCUSSION TITLE
How to perform a proximity search with Directus SDK?

### DISCUSSION THREAD

**Discussion by Yvan2XEro** [directus/directus#18115] [Archive]
Dear Directus Community,

I am building an advertisement website using Directus as a CMS. I have added a "location" field to my "advertisements" collection that stores the geographic coordinates of each ad as a JSON string. I now want to allow the user to search for ads near their current position, within a specified radius.

I am using the` @directus/sdk` package to perform queries in my frontend application. However, I have not been able to find any guidance in the Directus documentation on how to perform a proximity search using the SDK.

Can someone please provide guidance on how to construct a proximity search query with the Directus SDK, and how to integrate it into my frontend application?

**Comment by ctholho** [comment]
If you want to implement the proximity search yourself and perform it on the server, you need to use [custom endpoints](https://docs.directus.io/extensions/endpoints.html#custom-api-endpoints). Depending on the amount of data I would rely on the database to perform the actual computation for speed. [This stack overflow](https://gis.stackexchange.com/questions/21903/how-do-i-perform-a-proximity-search-with-postgis) might be a good starting point if you're using postgres.

If you have small amounts of data you can simply do it in your custom endpoint. See how it works in general [in this tutorial](https://javascript.plainenglish.io/javascript-algorithm-distance-between-points-7fe0026857e3).

If you're familiar with PHP, take a look into [this older question](https://github.com/directus/directus/discussions/2720), where they use an older version of directus.

If you have very little data (e.g. max 1000 items, though more is likely feasible as well) and you want to perform the calculations on your frontend, you can load all data with the SDK and then simply use the same principles as described in the tutorial.

**Comment by rijkvanzanten** [comment]
You should be able to search within a given bounding box with the `_intersects_bbox` filter operator. Not entirely the same as "within N miles of point Y", but should get you there 🙂 (It'd just search in a square rather than a circle)

**Reply by Yvan2XEro** [reply]
Thank you for the response, but I have a problem with using this filter. I'm getting a 400 error. I even tried using geojson, but it's possible that the syntax expected by Directus is particular. 
Here is my query in question and the calculation of the coordinates of the square:

![image](https://user-images.githubusercontent.com/63490400/231187551-d4234ddb-521e-4ebc-9919-e5d505dceed0.png)

Here is the error:

![image](https://user-images.githubusercontent.com/63490400/231188853-b5eacd73-09ae-4c8c-a953-55278266515c.png)

Thanks for help

**Reply by rijkvanzanten** [reply]
What's the output of "GeoJson.parse"?

**Reply by Yvan2XEro** [reply]
![image](https://user-images.githubusercontent.com/63490400/231257901-5095681f-6ec8-49cd-8edf-f9b580b7ffb7.png)


[Here are explanations](https://www.npmjs.com/package/geojson)

**Reply by rijkvanzanten** [reply]
Ah that looks like a feature collection of 4 points rather than a single object, can you try it as a single polygon?

**Comment by michaeltoohig** [comment]
If anyone was wondering about this as I found no documentation here is a working example. The catch is the `_intersects` filter expects a GeoJSON Feature of a Polygon not a typical BoundingBox. And as https://github.com/directus/directus/discussions/19329 states `_intersects` appears to be the same as `_intersects_bbox`.

```
const response = await directus.items('Land').readByQuery({
  limit: 100,
  fields: ['id', 'name', 'center', 'geometry'],
  filter: {
    center: {
      _intersects: {
      "type": "Feature",
      "properties": {},
      "geometry": {
        "coordinates": [
          [
            [
              -24.986437052474145,
              17.12242374698073
            ],
            [
              -24.986437052474145,
              17.120870396135288
            ],
            [
              -24.984886112693715,
              17.120870396135288
            ],
            [
              -24.984886112693715,
              17.12242374698073
            ],
            [
              -24.986437052474145,
              17.12242374698073
            ]
          ]
        ],
        "type": "Polygon"
      }
    },
  },
});
```
